E2E Test Utils: Improve durability of embedding matcher

aduth opened the pull request:
This pull request seeks to improve the implementation of `createEmbeddingMatcher` to be more durable to inputs. Specifically:

- Account for varying site permalink configuration, where the current implementation assumes the default permalink configuration (`?rest_route=`), and would fail with "pretty permalinks"
- Use `URL.searchParams` for the implementation of `parameterEquals`, to avoid edge cases like:
   - `https://url=why` (doesn't verify that it's part of the query string)
   - `https://example.com/?noturl=why` (doesn't verify the full parameter name)

This was implemented in testing #20481, though it didn't appear related to the issues being debugged there. So this is purely meant as a refactoring.

**Testing Instructions:**

End-to-end tests should pass:

```
npm run test-e2e
```

aduth committed:
E2E Test Utils: Improve durability of embedding matcher

gziolo reviewed (approved):
I assume that the failing e2e test on Travis is meant to be fixed with https://github.com/WordPress/gutenberg/pull/20481.

🚢

aduth commented:
> I assume that the failing e2e test on Travis is meant to be fixed with #20481.

It hopes to, though not entirely effective at least based on the build history there 😄 I might think to still merge it though, if it can at least reduce the number of failures.

aduth committed:
E2E Test Utils: Add newline between functions

aduth committed:
E2E Test Utils: Avoid normalizing path endings

(a) It's not consistent with preceding `parameterEquals` testing and (b) there's not variance here, for all requests via `getEmbedPreview` implementation

See: https://github.com/WordPress/gutenberg/blob/3848806ccc7b8a65778366e80af7ea63258538d4/packages/core-data/src/resolvers.js#L112
